File: packages/lynx/examples/native-capabilities/android/src/main/java/dev/octane/example/OctaneBadgeView.java
package dev.octane.example;

import android.content.Context;
import android.widget.TextView;
import com.lynx.tasm.behavior.LynxContext;
import com.lynx.tasm.behavior.LynxElement;
import com.lynx.tasm.behavior.LynxProp;
import com.lynx.tasm.behavior.ui.LynxUI;

/** Application-owned custom element; the host's Lynx registry owns it. */
@LynxElement(name = "octane-badge")
public final class OctaneBadgeView extends LynxUI<TextView> {
  public OctaneBadgeView(LynxContext context) {
    super(context);
  }

  @Override
  protected TextView createView(Context context) {
    return new TextView(context);
  }

  @LynxProp(name = "label")
  public void setLabel(String label) {
    if (mView != null) {
      mView.setText(label);
    }
  }
}

File: packages/lynx/examples/native-capabilities/ios/OctaneBadgeView.m
#import "OctaneBadgeView.h"
#import <Lynx/LynxPropsProcessor.h>

@LynxElement("octane-badge")
@implementation OctaneBadgeView

- (UILabel *)createView {
  return [[UILabel alloc] init];
}

LYNX_PROP_SETTER("label", setLabel, NSString *) {
  self.view.text = value;
}

@end
